external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -2265,3 +2275,57 @@ fn make_arg_types<S: std::fmt::Display, F: Fn() -> S>(
     }
     given
 }
+
+//**************************************************************************************************
+// Module-wide warnings
+//**************************************************************************************************
+
+/// Generates warnings for unused struct types and unused (private) functions.
+fn gen_unused_warnings(context: &mut Context, ident: ModuleIdent, mdef: &T::ModuleDefinition) {
+    if context.compiled_module_names.contains(&ident) {
+        // generate warnings only for modules compiled in this pass rather than for all modules
+        // including pre-compiled libraries for which we do not have the right type of data
+        // collected (as they are not actually compiled here)
+        context
+            .env
+            .add_warning_filter_scope(mdef.warning_filter.clone());
+
+        for (loc, name, fun) in &mdef.functions {
+            context
+                .env
+                .add_warning_filter_scope(fun.warning_filter.clone());
+            if !context.called_fns.contains(name)
+                && fun.entry.is_none()
+                && matches!(fun.visibility, Visibility::Internal)
+            {
+                // TODO: postponing handling of friend functions until we decide what to do with them
+                // vis-a-vis ideas around package-private
+                let msg = format!(
+                    "The non-'public', non-'entry' function '{name}' is never called. \
+                           Consider removing it."
```

Comment:
People have complained to me in the past about my passive aggressive compiler warnings like this, but I'm fine with it ;)

Reply:
I simply followed the guideline from the previous review :-)

Reply:
Haha sure, I'm just trying to say that I'm open to suggestion on the wording here